testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/AdminClientTest.java

```diff
@@ -192,6 +214,30 @@ public void scopedClientCredentialsAuthSuccess() throws Exception {
         }
     }
 
+    // A client secret in not necessary when authentication is
+    // performed via X.509 authorizer.
+    @Test
+    public void noClientSecretWithClientCredentialsAuthSuccess() throws Exception {
+        final RealmResource testRealm = adminClient.realm(realmName);
+
+        final String scopeName = "dummyScope";
+        String scopeId = createScope(testRealm, scopeName, KeycloakModelUtils.generateId());
+        testRealm.clients().get(x509ClientUUID).addOptionalClientScope(scopeId);
+
+        // with scope and no client secret
+        try (Keycloak adminClient = AdminClientUtil.
+            createMTlsAdminClientWithClientCredentialsWithoutSecret(realmName, x509ClientId, scopeName)) {
+            final AccessTokenResponse accessToken = adminClient.tokenManager().getAccessToken();
+            Assert.assertTrue(accessToken.getScope().contains(scopeName));
```

Comment:
Is it possible to test something here with the adminClient to doublecheck that it works and can be used to invoke endpoints?

Maybe just something like adding this line is sufficient:
```
Assert.assertNotNull(adminClient.realm(realmName).clientScopes().get(scopeId).toRepresentation());
```

Reply:
That is actually a very good catch. To make it possible, I had to create an additional user otherwise I don't get proper `realm-management:realm-admin` service account role (thus getting `403` all the time). Not sure, if that is the way it should work or whether there is easier way to configure service account. But apparently, this is what other tests rely on. I left a comment in the code because it is not something obvious.

Resolution: done